lib/srv/regular/sshserver.go

```diff
@@ -1420,6 +1420,23 @@ func (s *Server) dispatch(ch ssh.Channel, req *ssh.Request, ctx *srv.ServerConte
 		}
 	}
 
+	if ctx.JoinOnly {
```

Comment:
I don't love the duplication here, but I guess it's nice to keep the join-only logic isolated.

Maybe add a comment at the beginning of this block.

```
// Certs with a join-only principal can only use a
// subset of all the possible request types.
```